tests/test_precommit_setup.py

```
@@ -0,0 +1,197 @@
+        """Test that .pre-commit-config.yaml exists and contains valid YAML structure."""
+        # Arrange
+        repo_root = Path(__file__).parent.parent
+        config_file = repo_root / ".pre-commit-config.yaml"
+
+        # Act & Assert
+        assert config_file.exists(), (
+            f".pre-commit-config.yaml not found at {config_file}"
+        )
+
+        # Verify it's valid YAML
+        with open(config_file) as f:
```

Ruff (0.12.2)

`open()` should be replaced by `Path.open()`

(PTH123)